src/fldeff_flash.c

```diff
@@ -14,53 +14,54 @@
-static void sub_8137420(u8 taskId);
-static void sub_81374C4(u8 taskId);
-static void sub_813750C(u8 taskId);
-static void sub_8137574(u8 taskId);
-static void sub_81375A8(void);
-static void sub_81375BC(u8 taskId);
-static void sub_81375D8(u8 taskId);
-static void sub_8137678(u8 taskId);
-static void sub_81376DC(u8 taskId);
+static bool8 TryDoMapTransition(void);
+static void DoExitCaveTransition(void);
+static void Task_ExitCaveTransition1(u8 taskId);
+static void Task_ExitCaveTransition2(u8 taskId);
+static void Task_ExitCaveTransition3(u8 taskId);
+static void Task_ExitCaveTransition4(u8 taskId);
+static void Task_ExitCaveTransition5(u8 taskId);
+static void DoEnterCaveTransition(void);
+static void Task_EnterCaveTransition1(u8 taskId);
+static void Task_EnterCaveTransition2(u8 taskId);
+static void Task_EnterCaveTransition3(u8 taskId);
+static void Task_EnterCaveTransition4(u8 taskId);
 
 // rodata
-static const struct FlashStruct gUnknown_085B27C8[] =
+static const struct FlashStruct sTransitionTypes[] =
 {
-    {1, 4, 1, 0, sub_81375A8},
-    {2, 4, 1, 0, sub_81375A8},
-    {3, 4, 1, 0, sub_81375A8},
-    {5, 4, 1, 0, sub_81375A8},
-    {6, 4, 1, 0, sub_81375A8},
-    {7, 4, 1, 0, sub_81375A8},
-    {8, 4, 1, 0, sub_81375A8},
-    {9, 4, 1, 0, sub_81375A8},
-    {4, 1, 0, 1, sub_81373F0},
-    {4, 2, 0, 1, sub_81373F0},
-    {4, 3, 0, 1, sub_81373F0},
-    {4, 5, 0, 1, sub_81373F0},
-    {4, 6, 0, 1, sub_81373F0},
-    {4, 7, 0, 1, sub_81373F0},
-    {4, 8, 0, 1, sub_81373F0},
-    {4, 9, 0, 1, sub_81373F0},
-    {0, 0, 0, 0, NULL},
+    {MAP_TYPE_TOWN,        MAP_TYPE_UNDERGROUND,  TRUE, FALSE, DoEnterCaveTransition},
+    {MAP_TYPE_CITY,        MAP_TYPE_UNDERGROUND,  TRUE, FALSE, DoEnterCaveTransition},
+    {MAP_TYPE_ROUTE,       MAP_TYPE_UNDERGROUND,  TRUE, FALSE, DoEnterCaveTransition},
+    {MAP_TYPE_UNDERWATER,  MAP_TYPE_UNDERGROUND,  TRUE, FALSE, DoEnterCaveTransition},
+    {MAP_TYPE_OCEAN_ROUTE, MAP_TYPE_UNDERGROUND,  TRUE, FALSE, DoEnterCaveTransition},
+    {MAP_TYPE_UNUSED_2,    MAP_TYPE_UNDERGROUND,  TRUE, FALSE, DoEnterCaveTransition},
+    {MAP_TYPE_INDOOR,      MAP_TYPE_UNDERGROUND,  TRUE, FALSE, DoEnterCaveTransition},
+    {MAP_TYPE_SECRET_BASE, MAP_TYPE_UNDERGROUND,  TRUE, FALSE, DoEnterCaveTransition},
+    {MAP_TYPE_UNDERGROUND, MAP_TYPE_TOWN,        FALSE,  TRUE, DoExitCaveTransition},
+    {MAP_TYPE_UNDERGROUND, MAP_TYPE_CITY,        FALSE,  TRUE, DoExitCaveTransition},
+    {MAP_TYPE_UNDERGROUND, MAP_TYPE_ROUTE,       FALSE,  TRUE, DoExitCaveTransition},
+    {MAP_TYPE_UNDERGROUND, MAP_TYPE_UNDERWATER,  FALSE,  TRUE, DoExitCaveTransition},
+    {MAP_TYPE_UNDERGROUND, MAP_TYPE_OCEAN_ROUTE, FALSE,  TRUE, DoExitCaveTransition},
+    {MAP_TYPE_UNDERGROUND, MAP_TYPE_UNUSED_2,    FALSE,  TRUE, DoExitCaveTransition},
+    {MAP_TYPE_UNDERGROUND, MAP_TYPE_INDOOR,      FALSE,  TRUE, DoExitCaveTransition},
+    {MAP_TYPE_UNDERGROUND, MAP_TYPE_SECRET_BASE, FALSE,  TRUE, DoExitCaveTransition},
+    {0, 0, FALSE, FALSE, NULL},
```

Comment:
This last entry does not need to be explicitly initialized, as all the fields evaluate to 0.  Just use empty braces (`{}`).  The C standard states that uninitialized elements of a struct in ROM or static RAM will be filled with 0s.